gateway/resolver/arguments.go

```diff
@@ -129,3 +146,30 @@ func getBoolArg(args map[string]interface{}, key string, required bool) (bool, e
 func isResourceNamespaceScoped(resourceScope apiextensionsv1.ResourceScope) bool {
 	return resourceScope == apiextensionsv1.NamespaceScoped
 }
+
+func validateSortBy(items []unstructured.Unstructured, fieldPath string) (string, error) {
+	if len(items) == 0 {
+		return "", nil // No items to validate against, assume valid
+	}
+
+	sample := items[0]
+	segments := strings.Split(fieldPath, ".")
+
+	if _, found, err := unstructured.NestedString(sample.Object, segments...); err == nil && found {
```

Comment:
There is another way to solve it using graphql introspection, but this way is much more simpler and readable. I would keep it.